### PR TITLE
Feat/redux

### PR DESCRIPTION
**Pull request by sonishreyas**
# Change logs 
- Added redux to the code

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
code looks clean .good to be merged

File: src/backend/utils/authentication.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				};
+				localStorage.setItem("user", JSON.stringify(userData));
+				const newUserRef = doc(db, "users", userData.uid);
+				await setDoc(newUserRef, {
+					...userData,
+					followers: [],
+					following: [],
+					bookmarks: [],
+				});
+				// showToast("User registered successfully..", "success");
```
Comment (parthpandyappp) [inline src/backend/utils/authentication.js:74]: Remove this comment.

File: src/backend/utils/authentication.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+				console.log(userData);
```
Comment (parthpandyappp) [inline src/backend/utils/authentication.js:75]: Remove redundant log.

File: src/backend/utils/post-management.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			uploadTask.on(
+				"state_changed",
+				(snapshot) => {
+					// Get task progress, including the number of bytes uploaded and the total number of bytes to be uploaded
+					const progress =
+						(snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+					console.log("Upload is " + progress + "% done");
+					switch (snapshot.state) {
+						case "paused":
+							console.log("Upload is paused");
```
Comment (parthpandyappp) [inline src/backend/utils/post-management.js:39]: Here also the same, redundant logs.

File: src/backend/utils/post-management.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+							break;
+						case "running":
+							console.log("Upload is running");
```
Comment (parthpandyappp) [inline src/backend/utils/post-management.js:42]: Remove this too.

**Review by parthpandyappp — COMMENTED** [review]
Code looks clean to now. Found no breaking changes. I've commented on few line, look into it. There are some redundant logs just remove them and you're good to go for the merge :+1:

**Review by parthpandyappp — APPROVED** [review]

**Pull request merged**